utils/pao-profile.sh

```diff
@@ -0,0 +1,207 @@
+        reserved+=($cpu)
+      fi
+        count=$((count+1))
+    done
+
+    # add the remaining CPUs to reserved
+    num_cpus=${#cpus_1[@]}
+    count=0
+    max=$((($num_cpus -8) / 2))
+    for cpu in ${cpus_1[@]}; do
+      if [ $count -le $max ] ; then
+        # add the cpu to the reserved nodes
+        reserved+=($cpu)
+      fi
+      count=$((count+1))
+    done
+  fi
+
+
+  # numa node is 1
+  if [[ $numa_node == 1 ]]; then
+    # all cpus in cpus_1 - 2 for housekeeping go to isolated
+    log "Numa node is 1"
+    export SOCKET_MEMORY="0,1024"
+    num_cpus=${#cpus_1[@]}
+    count=0
+    max=$((($num_cpus -8) / 2))
+    #echo max is $max
+    max_isol=$((max - 2))
+    for cpu in ${cpus_1[@]}; do
+      if [ $count -le $max_isol ]; then
+        # add the cpu to the isolated nodes
+        isolated+=($cpu)
+      elif [ $count -gt $max_isol ] && [ $count -le $max ]; then
+        # add the cpu to the reserved nodes for housekeeping
+        reserved+=($cpu)
+      fi
+        count=$((count+1))
+    done
+
+    # add the remaining CPUs to reserved
+    num_cpus=${#cpus_0[@]}
+    count=0
+    max=$((($num_cpus -8) / 2))
+    #echo max: $max
+    for cpu in ${cpus_0[@]}; do
+      if [ $count -le $max ] ; then
+      # add the cpu to the reserved nodes
+      reserved+=($cpu)
+      fi
+      count=$((count+1))
+    done
+  fi
+
+  # templatize the perf profile and the sriov network node policy
+  reserved_string=$(echo ${reserved[@]} | sed -s 's/ /,/g')
+  isolated_string=$(echo ${isolated[@]} | sed -s 's/ /,/g')
+  export isolated_cpus=$isolated_string
+  export reserved_cpus=$reserved_string
+
+  # label the two nodes for the performance profile
+  log "Labeling -rt nodes"
+  for w in ${testpmd_workers[@]}; do
+    oc label node $w node-role.kubernetes.io/worker-rt="" --overwrite=true
+  done
+
+  # create the machineconfigpool
+  log "Create the MCP"
+  envsubst < $MCP | oc apply -f -
+  sleep 30
+  if [ $? -ne 0 ] ; then
+    log "Couldn't create the MCP, exiting!"
+    exit 1
+  fi
+
+  # add the label to the MCP pool 
+  log "Labeling the MCP"
+  oc label mcp worker-rt machineconfiguration.openshift.io/role=worker-rt --overwrite=true
+  if [ $? -ne 0 ] ; then
+    log "Couldn't label the MCP, exiting!"
+    exit 1
+  fi
+
+  # apply the performanceProfile
+  log "Applying the performanceProfile"
+  envsubst < $PFP | oc apply -f -
+  if [ $? -ne 0 ] ; then
+    log "Couldn't apply the performance profile, exiting!"
+    exit 1
+  fi
+
+  # We need to wait for the nodes with the perfProfile applied to to reboot
+  # this is a catchall approach, we sleep for 60 seconds and check the status of the nodes
+  # if they're ready, we'll continue. Should the performance profile require reboots, that will have
+  # started within the 60 seconds
+  log "Sleeping for 60 seconds"
+  sleep 60
+  readycount=$(oc get mcp worker-rt --no-headers | awk '{print $7}')
+  while [[ $readycount -ne 2 ]]; do
```

Comment:
There is a potentially infinite loop here if a node never comes back. Can we add in a timeout of some sort?